cmd/oklog/ingeststore.go

```diff
@@ -355,6 +358,17 @@ func runIngestStore(args []string) error {
 		},
 	}
 
+	recordRdr := ingest.NewDynamicRecordReader
```

Comment:
Hmm, this is picky, but both `ingest.RecordReader` and `func(io.Reader) ingest.RecordReader` are being called `recordReader`. May I suggest:

```go
package ingest

type RecordReaderFactory func(io.Reader) RecordReader

func NewDynamicRecordReader(r io.Reader) RecordReader { 
    // ... 
}

func StaticRecordReader(topic []byte) RecordReaderFactory {
    return func(r io.Reader) RecordReader { 
        // ...
    }
}
```

```go
package main

var rfac ingest.RecordReaderFactory
{
    switch {
    case *topicMode == topicModeDynamic:
        rfac = ingest.NewDynamicRecordReader
    case *topicMode == topicModeStatic && ingest.IsValidTopic(*topic):
        rfac = ingest.StaticRecordReader(*topic)
    case *topicMode == topicModeStatic && !ingest.IsValidTopic(*topic):
        // log invalid topic and exit
    default:
        // log invalid topic mode and exit
    }
}
```

Reply:
I take that you'd like `StaticRecordReader` to panic on an invalid topic?